### PR TITLE
Sellers/user name

### PR DESCRIPTION
**Pull request by Valseer**
Related Issue #85 

Changes:
- User Name Added to bar
- User stored in session now
- Put both User Name and Sign out in Popover

Breaking Changes:
-

### CODE REVIEW

File: client/src/components/UserName.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+
+export interface UserNameProps {
+    isLoggedIn: boolean;
+    userName: string;
```
Comment (djrenren) [inline client/src/components/UserName.tsx]: I'd make this optional. In theory we could detect `isLoggedIn` based on the presence of the userName string.

File: client/src/components/UserName.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    error?: string;
```
Comment (djrenren) [inline client/src/components/UserName.tsx]: You don't use this param.

File: client/src/components/UserName.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}
+const UserName = (props: UserNameProps) => {
+    let name= <div> {props.userName} </div>;
+    let noName = <div />;
+    return (
```
Comment (djrenren) [inline client/src/components/UserName.tsx]: Why not?
```javascript

const UserName = (props: UserNameProps) =>
    <div style={{marginTop: '6px'}}>
            {props.userName}
    </div>
```
Also 6px seems odd...

File: client/src/components/UserName.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        <div style={{marginTop: '6px'}}>
+            {props.isLoggedIn ? name : noName }
+        </div>
+    );
+};
+export default UserName;
```
Comment (djrenren) [inline client/src/components/UserName.tsx]: newline pls

File: client/src/components/AppBarRight.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    FlatButton
+} from 'material-ui';
+
+export interface AppBarRightProps {
+    isLoggedIn: boolean;
+    userName?: string;
+    onLogoutClick?: () => void;
+    error?: string;
+}
+const AppBarRight = (props: AppBarRightProps) => {
```
Comment (djrenren) [inline client/src/components/AppBarRight.tsx]: Could we give this a more semantic name?

File: client/src/components/AppBarRight.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+import {
+    FlatButton
+} from 'material-ui';
+
+export interface AppBarRightProps {
+    isLoggedIn: boolean;
+    userName?: string;
+    onLogoutClick?: () => void;
+    error?: string;
```
Comment (djrenren) [inline client/src/components/AppBarRight.tsx]: You don't use `error`

File: client/src/components/AppBarRight.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+
+import {
+    FlatButton
+} from 'material-ui';
+
+export interface AppBarRightProps {
+    isLoggedIn: boolean;
+    userName?: string;
```
Comment (djrenren) [inline client/src/components/AppBarRight.tsx]: Infer `isLoggedIn` from userName

File: client/src/reducers/sessionReducer.ts [hunk 8]
```diff
@@ -1,16 +1,25 @@
...
     pending?: boolean;
+    user?: User;
     error?: string;
 }
 
+export interface User {
+    id: number;
+    name: string;
+    email: string;
+    bDay: string;
```
Comment (djrenren) [inline client/src/reducers/sessionReducer.ts]: we should probably store this as a Date

**Review by djrenren — DISMISSED** [review]

File: client/src/components/AppBarRight.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    error?: string;
+}
+const AppBarRight = (props: AppBarRightProps) => {
+    let buttons = <FlatButton onClick={props.onLogoutClick}
+        style={{ color: 'white', float:'right'}}
+        label='Logout' />;
+    let name = <div style= {{ color: 'white', float: 'left', marginTop: '7px'}}> {props.userName} |</div>;
+    let right = <div> {name}  {buttons} </div>;
+    let notButtons = <div />;
+    return (
```
Comment (djrenren) [inline client/src/components/AppBarRight.tsx]:
```javascript
const style = { color: 'white', float: 'left', marginTop: '7px'};
let loggedIn = (
    <div style={{marginTop: '6px'}}>
        <div style = {style}> {props.userName} |</div>
        <FlatButton onClick={props.onLogoutClick}
                            style={buttonStyle}
                            lable='Logout' />
    </div>
);

return props.userName ? loggedIn : <div/>
```

**Review by 7imbrook — APPROVED** [review]

**Pull request merged**